posthog/api/event.py

```diff
@@ -261,274 +256,23 @@ def values(self, request: request.Request) -> response.Response:
 
         return response.Response([{"name": convert_property_value(value.value)} for value in values])
 
-    def _handle_compared(self, date_filter: Dict[str, datetime]) -> QuerySet:
-        date_from, date_to = get_compare_period_dates(date_filter["timestamp__gte"], date_filter["timestamp__lte"])
-        date_filter["timestamp__gte"] = date_from
-        date_filter["timestamp__lte"] = date_to
-        compared_events = self.get_queryset().filter(**date_filter)
-        return compared_events
-
-    def _convert_to_comparison(self, trend_entity: List[Dict[str, Any]], label: str) -> List[Dict[str, Any]]:
-        for entity in trend_entity:
-            days = [i for i in range(len(entity["days"]))]
-            labels = ["{} {}".format("Day", i) for i in range(len(entity["labels"]))]
-            entity.update(
-                {
-                    "labels": labels,
-                    "days": days,
-                    "chartLabel": "{} - {}".format(entity["label"], label),
-                    "dates": entity["days"],
-                    "compare": True,
-                }
-            )
-        return trend_entity
-
     @action(methods=["GET"], detail=False)
     def sessions(self, request: request.Request) -> response.Response:
         team = self.request.user.team_set.get()
         session_type = self.request.GET.get("session")
 
-        date_filter = request_to_date_query(request.GET.dict(), exact=True)
-        if not date_filter.get("timestamp__gte"):
-            date_filter["timestamp__gte"] = (
-                Event.objects.filter(team=team)
-                .order_by("timestamp")[0]
-                .timestamp.replace(hour=0, minute=0, second=0, microsecond=0)
+        filter = Filter(request=request)
+        result: Dict[str, Any] = {
+            "result": Sessions().run(
+                filter, team, session_type=self.request.GET.get("session"), offset=self.request.GET.get("offset")
             )
-
-        if not date_filter.get("timestamp__lte"):
-            date_filter["timestamp__lte"] = now()
-
-        events = self.get_queryset()
```

Comment:
property filtering seems to have no effect now. originally it was wrapped into the processing for `self.get_queryset()` and now i don't believe it's handled anywhere in sessions

Reply:
Fixed